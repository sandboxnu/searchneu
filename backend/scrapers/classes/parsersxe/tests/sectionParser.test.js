import SectionParser from '../sectionParser';
import data from './data/sectionParser.data';

describe('sectionParser', () => {
  const chem2311Section = SectionParser.parseSectionFromSearchResult(data.chem2311);
  const cs2500Section = SectionParser.parseSectionFromSearchResult(data.cs2500);
  it('should match snapshot', () => {
    // Snapshot test gives full coverage, but other tests also exist to clearly spec things out
    expect(chem2311Section).toMatchSnapshot();
    expect(cs2500Section).toMatchSnapshot();
  });

  it('should detect Honors', () => {
    expect(chem2311Section.honors).toBeTruthy();
    expect(cs2500Section.honors).toBeFalsy();
  });

  it('should detect campus', () => {
    expect(chem2311Section.campus).toBe('Online');
    expect(cs2500Section.campus).toBe('Boston');
  });
});
