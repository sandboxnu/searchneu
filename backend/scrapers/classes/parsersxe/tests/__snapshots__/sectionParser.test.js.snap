// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sectionParser should match snapshot 1`] = `
Object {
  "campus": "Online",
  "classId": "2311",
  "classType": "Lecture",
  "crn": "10259",
  "honors": true,
  "host": "neu.edu",
  "meetings": Array [
    Object {
      "endDate": 18234,
      "startDate": 18143,
      "times": Object {
        "1": Array [
          Object {
            "end": 52800,
            "start": 48900,
          },
        ],
        "3": Array [
          Object {
            "end": 52800,
            "start": 48900,
          },
        ],
        "4": Array [
          Object {
            "end": 52800,
            "start": 48900,
          },
        ],
      },
      "type": "Class",
      "where": "Shillman Hall 220",
    },
    Object {
      "endDate": 18243,
      "startDate": 18236,
      "times": Object {},
      "type": "Final Exam",
      "where": "TBA",
    },
  ],
  "profs": Array [
    "Oyindasola Oyelaran",
  ],
  "seatsCapacity": 76,
  "seatsRemaining": 36,
  "subject": "CHEM",
  "termId": "202010",
  "url": "https://wl11gp.neu.edu/udcprod8/bwckschd.p_disp_detail_sched?term_in=202010&crn_in=10259",
  "waitCapacity": 0,
  "waitRemaining": 0,
}
`;

exports[`sectionParser should match snapshot 2`] = `
Object {
  "campus": "Boston",
  "classId": "2500",
  "classType": "Lecture",
  "crn": "30340",
  "honors": false,
  "host": "neu.edu",
  "meetings": Array [
    Object {
      "endDate": 18003,
      "startDate": 17903,
      "times": Object {
        "1": Array [
          Object {
            "end": 37200,
            "start": 33300,
          },
        ],
        "3": Array [
          Object {
            "end": 37200,
            "start": 33300,
          },
        ],
        "4": Array [
          Object {
            "end": 37200,
            "start": 33300,
          },
        ],
      },
      "type": "Class",
      "where": "East Village 024",
    },
  ],
  "profs": Array [
    "Alan Mislove",
  ],
  "seatsCapacity": 75,
  "seatsRemaining": 4,
  "subject": "CS",
  "termId": "201930",
  "url": "https://wl11gp.neu.edu/udcprod8/bwckschd.p_disp_detail_sched?term_in=201930&crn_in=30340",
  "waitCapacity": 0,
  "waitRemaining": 0,
}
`;
